data/crac/crac-impl/src/main/java/com/powsybl/openrao/data/cracimpl/OnConstraintAdderImpl.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.openrao.data.cracimpl;
+
+import com.powsybl.openrao.commons.OpenRaoException;
+import com.powsybl.openrao.data.cracapi.Instant;
+import com.powsybl.openrao.data.cracapi.cnec.Cnec;
+import com.powsybl.openrao.data.cracapi.usagerule.OnConstraint;
+import com.powsybl.openrao.data.cracapi.usagerule.OnConstraintAdder;
+import com.powsybl.openrao.data.cracapi.usagerule.UsageMethod;
+
+import java.util.Objects;
+
+import static com.powsybl.openrao.data.cracimpl.AdderUtils.assertAttributeNotNull;
+
+/**
+ * @author Thomas Bouquet <thomas.bouquet at rte-france.com>
+ */
+public class OnConstraintAdderImpl<T extends AbstractRemedialActionAdder<T>, J extends Cnec<?>> implements OnConstraintAdder<T, J> {
```

Comment:
Is there a logic for choosing J as the second generic type name?
According to https://docs.oracle.com/javase/tutorial/java/generics/types.html, S would be a better choice if there's no logic behind the J choice.

The most commonly used type parameter names are:
    E - Element (used extensively by the Java Collections Framework)
    K - Key
    N - Number
    T - Type
    V - Value
    S,U,V etc. - 2nd, 3rd, 4th types

Reply:
Using J was completely random, I'll update this :ok_hand: